### ISSUE TITLE
[C# operations] treat C# keywords

### ISSUE THREAD

**Issue by fialazd2** [dotansimha/graphql-code-generator#5556]
**Describe the bug**
C# operations can generate fields which are in collision with C# keywords (i.e. "case"). We use "case" in our schema.

**To Reproduce**
1. Generate C# operations.

1. My GraphQL schema:

```graphql
scalar Date

schema {
  query: Query
}

type Query {
  me: case!
  case(id: ID!): case
  allcases: [case]
  search(term: String!): [SearchResult!]!
  myChats: [Chat!]!
}

enum Role {
  case,
  ADMIN,
}

interface Node {
  id: ID!
}

union SearchResult = case | Chat | ChatMessage

type case implements Node {
  id: ID!
  casename: String!
  email: String!
  role: Role!
}

type Chat implements Node {
  id: ID!
  cases: [case!]!
  messages: [ChatMessage!]!
}

type ChatMessage implements Node {
  id: ID!
  content: String!
  time: Date!
  case: case!
}
```

2. My GraphQL operations:

```graphql
query findUser($caseId: ID!) {
  case(id: $caseId) {
    ...UserFields
  }
}

fragment UserFields on case {
  id
  casename
  role
}
```

3. My `codegen.yml` config file:

```yml
generates:
  src/main/c-sharp/my-org/my-app/Operations.cs:
    plugins:
      - c-sharp-operations
    config:
      typesafeOperation: true
```

**Expected behavior**
C# keywords like "case" should be prefixed with @ => `@case`

**Environment:**

- OS: 
- `@graphql-codegen/c-sharp-operations`: 
- NodeJS: 

**Additional context**

**Comment by ardatan** [comment]
Thanks for reporting the issue! You can use `namingConvention` or `prefix` to make it safe variable :)

**Comment by fialazd2** [comment]
Hi, thank you for your reply. Which `prefix` do you mean, please?

I tried:
fragmentVariablePrefix, typesPrefix, documentVariablePrefix

but none of then worked for generated property:
`        [JsonProperty("case")]
        public caseSelection case { get; set; }`

**Comment by HerrNiklasRaab** [comment]
Can reproduce this as well.

**Comment by Urigo** [comment]
Hi @HerrNiklasRaab and thank you for the report!

Sorry but I'm not adding a lot here but just labeling it according to our new Contribution Guide and [issue flow](https://github.com/the-guild-org/Stack/blob/master/CONTRIBUTING.md#a-typical-contributor-workflow).

It seems already got into `stage 0`.

Now in order to advance to `stage 1` we'll need a simple reproduction, maybe in code-sandbox? 

Later to progress to `stage 2`, a failing test would be needed, would be great if someone could help progress the issues through the [stages](https://github.com/the-guild-org/Stack/blob/master/CONTRIBUTING.md#a-typical-contributor-workflow).

Thank you and sorry that this comment is not a complete solution (yet).

**Comment by huderlem** [comment]
Hi, I ran into this bug today when setting up C# codegen.  A simple way to reproduce the bug is within the demo page on https://www.graphql-code-generator.com/.  Below is a screenshot of how the `c-sharp-operations` plugin does not prefix reserved keywords with the `@` character.
![image](https://user-images.githubusercontent.com/2229288/129813822-0b16b850-00b8-4ed5-b32d-30e265adc243.png)

I poked around the code and discovered that the [`CSharpOperationsVisitor`](CSharpOperationsVisitor) simply doesn't contain the keyword-related logic, like its sister [`CSharpResolversVisitor`](CSharpResolversVisitor ).

I intend to move this bug forward by adding failing tests and implementing the solution.
